### PR TITLE
filesystem: report estimated partition size minimums

### PR DESCRIPTION
**Pull request by dbungert**

### CODE REVIEW

File: subiquity/common/types.py [hunk 1]
```diff
@@ -268,6 +268,7 @@ class Partition:
     boot: Optional[bool] = None
     os: Optional[OsProber] = None
     offset: Optional[int] = None
+    estimated_min_size: Optional[int] = -1
```
Comment (ogayot) [inline subiquity/common/types.py:271]: If I understand correctly, this should never be None but I might have overlooked something

Comment (dbungert) [reply subiquity/common/types.py:271]: The Partition object is a bit different as it has two major use cases
* Originating from Subiquity, in which case we're going to give all the fields reasonable values.
* Originating from a client, which is only going to fill out a few fields.

While we could ask clients to send a slightly different object with just the required fields, doing it this way allows the client to get a Partition from a previous call, modify it, then send it back.
Also, the serializer objects if there are extra fields.

Comment (ogayot) [reply subiquity/common/types.py:271]: Thanks for the details! I understand the use-cases better.
Would None be a valid value when the Partition objects originates from Subiquity? or when it originates from the client?
I was expecting the value -1 to be set by the deserializer if `estimated_min_size` is absent from the request, is that incorrect?

**Review by mwhudson — COMMENTED** [review]
I think you should add or adapt an API test to exercise this?

**Comment by dbungert** [comment]
> I think you should add or adapt an API test to exercise this?

Done

File: subiquity/models/filesystem.py [hunk 2]
```diff
@@ -702,6 +702,16 @@ def boot(self):
...
 
+    @property
+    def estimated_min_size(self):
+        fs_data = self._m._probe_data.get('filesystem', {}).get(self._path())
+        if fs_data is None:
+            return -1
+        val = fs_data.get('ESTIMATED_MIN_SIZE')
+        if val == 0:
+            return self.device.alignment_data().part_align
+        return val
```
Comment (mwhudson) [inline subiquity/models/filesystem.py:713]: I don't know how likely this is to be a problem in practice but this should round the probert supplied value to part_align, and also round any value smaller than min_gap_size to min_gap_size (I'm now getting side-tracked by the design of PartitionAlignmentData -- how like is it that part_align and min_gap_size will be different? Why is it called min_gap_size and not min_part_size?)

**Review by mwhudson — CHANGES_REQUESTED** [review]
Sorry, I found something else to quibble over.

File: subiquity/models/filesystem.py (hunk 2 above)
Comment (mwhudson) [reply subiquity/models/filesystem.py]: Er yes, I guess min(align_up(ESTIMATED_MIN_SIZE), actual_size) or something.

**Review by mwhudson — APPROVED** [review]

File: subiquity/models/filesystem.py [hunk 3]
```diff
@@ -702,6 +702,18 @@ def boot(self):
...
+    def estimated_min_size(self):
+        fs_data = self._m._probe_data.get('filesystem', {}).get(self._path())
+        if fs_data is None:
+            return -1
+        val = fs_data.get('ESTIMATED_MIN_SIZE', -1)
+        if val == 0:
+            return self.device.alignment_data().part_align
+        if val == -1:
+            return -1
+        return align_up(val)
```
[the remark this answers is not in the document]
Comment (dbungert) [reply subiquity/models/filesystem.py:715]: ty

**Pull request merged**